packages/core/src/dynamoDb/utils/editItem.ts

```diff
@@ -0,0 +1,158 @@
+    tempFilePath: string,
+    selectedRow: RowData,
+    { tableName, regionCode, tableSchema }: { tableName: string; regionCode: string; tableSchema: TableSchema }
+) {
+    try {
+        const updatedContent = JSON.parse(fs.readFileSync(tempFilePath, 'utf-8'))
+        const response = await updateDynamoDbItem(updatedContent, selectedRow, { tableName, regionCode, tableSchema })
+        if (response) {
+            void vscode.window.showInformationMessage('Item updated successfully in DynamoDB')
+        }
+    } catch (error) {
+        void vscode.window.showErrorMessage(`Failed to update item in DynamoDB: ${(error as any).message}`)
+    }
+}
+async function updateDynamoDbItem(
+    updatedItem: RowData,
+    currentItem: RowData,
+    { tableName, regionCode, tableSchema }: { tableName: string; regionCode: string; tableSchema: TableSchema },
+    client: DynamoDbClient = new DynamoDbClient(regionCode)
+) {
+    const partitionKey = tableSchema.partitionKey.name
+    const sortKey = tableSchema.sortKey?.name
+
+    if (updatedItem[partitionKey] !== currentItem[partitionKey]) {
+        void vscode.window.showErrorMessage(`Partition key '${partitionKey}' cannot be updated.`)
+        return
+    }
+
+    if (sortKey && updatedItem[sortKey] !== currentItem[sortKey]) {
+        void vscode.window.showErrorMessage(`Sort key '${sortKey}' cannot be updated.`)
+        return
+    }
+
+    if (!compareRowDataItems(updatedItem, currentItem)) {
+        return
+    }
+    const itemWithDataType: GetItemOutput = await getItem(client, currentItem, tableName, tableSchema)
+    const expressionAttributeValues = getExpressionAttributeValues(updatedItem, itemWithDataType, tableSchema)
+    const updateRequest: UpdateItemInput = {
+        TableName: tableName,
+        Key: getKeyObject(currentItem, tableSchema),
+        AttributeUpdates: expressionAttributeValues,
+    }
+    void telemetry.dynamodb_edit.run(async () => {
+        telemetry.record({ action: 'user' })
+        const response = await client.updateItem(updateRequest)
+        if (response) {
+            telemetry.record({ action: 'success' })
+        }
+        return response
+    })
+    return undefined
+}
+
+function getExpressionAttributeValues(updatedItem: RowData, currentItem: GetItemOutput, tableSchema: TableSchema) {
+    if (!currentItem.Item) {
+        return
+    }
+    const keys = Object.keys(updatedItem)
+    const expressionAttributeValues: AttributeUpdates = {}
+
+    for (const key of keys) {
+        const attributeMap = getAttributeValue(currentItem.Item[key])
+
+        if (attributeMap && updatedItem[key] !== attributeMap.value) {
+            expressionAttributeValues[key] = {
+                Action: 'PUT',
+                Value: {
+                    [attributeMap.key]: updatedItem[key],
+                },
+            }
+        }
+    }
+    return expressionAttributeValues
+}
+
+function compareRowDataItems(updatedItem: RowData, currentItem: RowData) {
```

Comment:
I wonder if we updated this function name we could make this more clear. Maybe something like areRowDataKeysEqual? Because I think its just comparing the keys